docs/reference/testing_api.md

```diff
@@ -0,0 +1,9 @@
+<!-- Autogenerated by reference.py -->
+
+# Testing API
```

Comment:
I also added a separate section to the API reference. So that the public API meant for testing is separate from the rest of the API reference.

<img width="1226" alt="Screenshot 2025-03-01 at 13 28 31" src="https://github.com/user-attachments/assets/dc4393ba-8596-450c-a371-322e75351ff9" />